### PR TITLE
Extension api improvements

### PR DESCRIPTION
**Pull request by fm-andres-rangel**

### CODE REVIEW

File: api/include/extractor/api.h [hunk 1]
```diff
@@ -360,6 +380,13 @@ struct extractor_api_v1 {
                                     ...);
   fm_type_decl_cp (*type_type_get)(fm_type_sys_t *ts);
   const char *(*type_sys_errmsg)(fm_type_sys_t *ts);
+  void (*module_type_add)(struct fm_comp_sys_module *mod,
```
Comment (maxim-trokhimtchouk) [inline api/include/extractor/api.h:383]: Doesn't make sense

File: api/include/extractor/api.h [hunk 2]
```diff
@@ -360,6 +380,13 @@ struct extractor_api_v1 {
...
+                          const fm_comp_def_t *def, fmc_error_t **error);
+  void (*module_type_add_func)(fm_comp_sys_module_init_func,
+                               struct fm_comp_sys_module *mod,
+                               const fm_comp_def_t *def, fmc_error_t **error);
```
Comment (maxim-trokhimtchouk) [inline api/include/extractor/api.h:387]: Doesn't make sense either

File: include/extractor/comp_sys.h [hunk 3]
```diff
@@ -143,8 +143,19 @@ FMMODFUNC const char *fm_comp_sys_error_msg(fm_comp_sys_t *s);
...
  * @brief Load extension module
  */
-FMMODFUNC bool fm_comp_sys_ext_load(fm_comp_sys_t *, const char *name,
-                                    const char *path);
+FMMODFUNC bool fm_comp_sys_ext_load(fm_comp_sys_t *, const char *name);
+
+/**
+ * @brief Load default search path for modules
+ */
+FMMODFUNC void fm_comp_sys_paths_set_default(struct fm_comp_sys *sys,
```
Comment (maxim-trokhimtchouk) [inline include/extractor/comp_sys.h:151]: need the default one

File: python/src/sys_def.hpp [hunk 4]
```diff
@@ -45,11 +45,17 @@ ExtractorSystem *ExtractorSystem_lazy(ExtractorSystem *obj) {
       free(errmsg);
       return nullptr;
     }
+    fmc_error_t *error;
```
Comment (maxim-trokhimtchouk) [inline python/src/sys_def.hpp:48]: No lazy system creation. Just create one in the extractor module

File: api/include/extractor/api.h [hunk 5]
```diff
@@ -228,6 +229,25 @@ typedef struct fm_call_ctx {
...
+  struct fm_comp_sys *sys; // the system that owns the module
+  fmc_ext_t handle;        // module handle. Return of dlopen()
+  char *name;              // module name (e.g. "fmtron")
+  char *file;              // file full path of the library
+  struct fm_comp_sys_module *next, *prev;
+};
+
+struct extractor_api_v1;
+
+typedef void (*fm_comp_sys_module_init_func)(struct extractor_api_v1 *,
```
Comment (maxim-trokhimtchouk) [inline api/include/extractor/api.h:247]: should be fm_comp_sys_module_init_v1

File: api/include/extractor/api.h [hunk 6]
```diff
@@ -228,6 +229,25 @@ typedef struct fm_call_ctx {
...
+                                             struct fm_comp_sys_module *,
```
Comment (maxim-trokhimtchouk) [inline api/include/extractor/api.h:248]: fm_comp_sys_t *

**Review by maxim-trokhimtchouk — CHANGES_REQUESTED** [review]

File: python/src/sys_def.hpp [hunk 7]
```diff
@@ -273,7 +307,30 @@ PyObject *ExtractorSystem_new() {
...
+  fmc_error_t *error;
+  fm_comp_sys_paths_set_default(self->sys, &error);
+  if (error) {
+    PyErr_SetString(PyExc_RuntimeError, "Unable to set default search path");
+    return nullptr;
+  }
+  fm_comp_sys_std_comp(self->sys);
+  fm_comp_sys_py_comp(self->sys);
+  self->to_delete = true;
+  for (auto &&comp : self->custom) {
```
Comment (maxim-trokhimtchouk) [inline python/src/sys_def.hpp]: can be removed

**Review by maxim-trokhimtchouk — APPROVED** [review]

**Review by maxim-trokhimtchouk — APPROVED** [review]

File: test/ext_exe.cpp [hunk 8]
```diff
@@ -113,7 +112,5 @@ TEST(ext_lib, stream) {
 
 GTEST_API_ int main(int argc, char **argv) {
   testing::InitGoogleTest(&argc, argv);
-  src_dir = argv[1];
-  lib_path = argv[2];
   return RUN_ALL_TESTS();
-}
+}
```
Comment (fm-fede-ravchina) [inline test/ext_exe.cpp:116]: end of line is missing

File: src/comp_sys.cpp [hunk 9]
```diff
@@ -64,6 +98,130 @@ void fm_comp_sys_cleanup(fm_comp_sys_t *s) {
...
+  struct fm_comp_sys_module *modtmp;
+  DL_FOREACH_SAFE(modhead, mod, modtmp) {
+    fm_comp_sys_module_destroy(mod);
+    free(mod);
+  }
+  s->modules = NULL;
+}
+
+static struct fm_comp_sys_module *
+mod_load(struct fm_comp_sys *sys, const char *dir, const char *modstr,
```
Comment (fm-fede-ravchina) [inline src/comp_sys.cpp:123]: shouldn't we reuse extension.h implementation, just like conveyor market data extension does? This logic was copied from yamal before we decided to separate the generic extension logic from the component API

**Review by fm-fede-ravchina — APPROVED** [review]

**Review by fm-fede-ravchina — CHANGES_REQUESTED** [review]
I've missclicked approve

**Review by maxim-trokhimtchouk — APPROVED** [review]

File: src/comp_sys.cpp (hunk 9 above)
Comment (maxim-trokhimtchouk) [reply src/comp_sys.cpp:123]: Yes, we should. I was confused. I thought that you discussed this with Andres.

**Review by fm-fede-ravchina — APPROVED** [review]

**Review by maxim-trokhimtchouk — APPROVED** [review]

**Pull request merged**